pkg/plugin/logrus_logger.go

```diff
@@ -0,0 +1,130 @@
+package plugin
+
+import (
+	"log"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/sirupsen/logrus"
+)
+
+// logrusLogger implements the hclog.Logger interface and
+// delegates all calls to a logrus logger.
+type logrusLogger struct {
+	impl  logrus.FieldLogger
+	level logrus.Level
+	name  string
+}
+
+// args are alternating key, value pairs, where the keys
+// are expected to be strings, and values can be any type.
+func argsToFields(args ...interface{}) logrus.Fields {
+	var (
```

Comment:
I implemented something yesterday like this:
```go
l := someLogrusLogger
for i := 0; i < len(fields); i += 2 {
  l = l.WithField(fmt.Sprintf("%v", fields[i]), fields[i+1])
}
```

That obviously isn't super efficient for large varargs. We could instead do something like
```go
fields := make(logrus.Fields, len(args)/2)
for i := 0; i < len(args); i += 2 {
  fields[args[i]] = args[i+1]
}
```

Reply:
yeah, i was getting tired when i wrote this code..

Reply:
updated this func.